xarray/plot/plot.py

```diff
@@ -449,6 +450,21 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
         if imshow_rgb:
             # Don't add a colorbar when showing an image with explicit colors
             add_colorbar = False
+            # Calculate vmin and vmax automatically for `robust=True`
+            if robust:
+                vmin_tmp, vmax_tmp = darray.quantile(
+                    q=[ROBUST_PERCENTILE, 1 - ROBUST_PERCENTILE])
+                vmin = vmin if vmin is not None else vmin_tmp
+                vmax = vmax if vmax is not None else vmax_tmp
+                del vmin_tmp, vmax__tmp
+                robust=False
+            # Scale interval [vmin .. vmax] to [0 .. 1] and clip to bounds
+            if vmin is not None or vmax is not None:
+                vmin = vmin if vmin is not None else darray.min()
+                vmax = vmax if vmax is not None else darray.max()
+                darray = (darray.astype(np.float32) - vmin) / (vmax - vmin)
```

Comment:
We ran into dynamic range issues w/ `uint32` being converted to `float32`, particularly as some people use an extreme high value as a bad-data flag.  https://github.com/matplotlib/matplotlib/pull/10133